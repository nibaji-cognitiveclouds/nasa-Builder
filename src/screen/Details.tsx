/** @format */

import axios from "axios";
import React from "react";
import { ActivityIndicator, Text, View } from "react-native";

const Details: React.FC = (prop: any) => {
	const [loading, setLoading] = React.useState(false);
	const [asteroid, setAsteroid] = React.useState({});
	const [noData, setNoData] = React.useState(false);

	function getData() {
		axios
			.get(
				`https://api.nasa.gov/neo/rest/v1/neo/${prop.route.params.id}?api_key=cgFoDzaE9hdKE9UousiEPaX0V0B8GaO0ICUOMexs`
			)
			.then((res) => {
				setAsteroid(res.data);
			})
			.catch(() => {
				setNoData(true);
			})
			.finally(() => setLoading(false));
	}

	React.useEffect(() => {
		setLoading(true);
		getData();
	}, []);

	return (
		<View
			style={{
				margin: 20,
				justifyContent: "center",
				flex: 1,
			}}
		>
			{loading ? (
				<ActivityIndicator />
			) : (
				<View
					style={{
						backgroundColor: "rgba(0,0,0,.2)",
						padding: 10,
						borderRadius: 10,
					}}
				>
					{noData ? (
						<Text>No asteroid found with that ID</Text>
					) : (
						<View>
							<Text>Name: {asteroid.name}</Text>
							<Text>NASA JPL URL: {asteroid.nasa_jpl_url}</Text>
							<Text>
								Is Potentially Hazardous:
								{asteroid.is_potentially_hazardous_asteroid ? " Yes" : " No"}
							</Text>
						</View>
					)}
				</View>
			)}
		</View>
	);
};

export default Details;
