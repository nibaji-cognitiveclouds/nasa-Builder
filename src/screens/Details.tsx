/** @format */

import axios from "axios";
import React, { FC, useEffect, useState } from "react";
import { ActivityIndicator, SafeAreaView, Text, View } from "react-native";
import { styles } from "../styles/screens";
import { nasaItem } from "../types/response";
import { detailsProp } from "../types/screens";

const Details: FC<detailsProp> = (props) => {
	const [data, setData] = useState<nasaItem>();
	const [loading, setLoading] = useState<boolean>(false);
	const [error, setError] = useState<boolean>(false);

	useEffect(() => {
		setLoading(true);
		axios
			.get(
				`https://api.nasa.gov/neo/rest/v1/neo/${props.route.params.id}?api_key=WaBk2sqO5hRFhZAegMGFcdSHsiHk7gFaLczNk3B9`
			)
			.then((res) => {
				setData(res.data);
			})
			.catch(() => {
				setError(true);
			})
			.finally(() => {
				setLoading(false);
			});
	}, []);

	return (
		<SafeAreaView testID="details" style={styles.container}>
			{loading ? (
				<ActivityIndicator color={"red"} size={20} />
			) : error ? (
				<Text>Something Went wrong!</Text>
			) : (
				<View>
					<Text>Name: {data?.name}</Text>
					<Text>Nasa JPl URL: {data?.nasa_jpl_url}</Text>
					<Text>
						Is potentially hazardous:{" "}
						{data?.is_potentially_hazardous_asteroid ? "yes" : "no"}
					</Text>
				</View>
			)}
		</SafeAreaView>
	);
};

export default Details;
