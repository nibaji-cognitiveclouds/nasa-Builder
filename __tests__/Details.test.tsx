/** @format */

import { render } from "@testing-library/react-native";
import React from "react";
import Details from "../src/screens/Details";

const props = {
	params: {
		id: "1234",
	},
};

describe("Details", () => {
	jest.useFakeTimers();
	it("renders without crash", () => {
		render(<Details route={props} />);
	});

	it("matches snapshot", () => {
		const tree = render(<Details route={props} />);
		expect(tree.toJSON()).toMatchSnapshot();
	});

	it("has component with req testID", () => {
		const tree = render(<Details route={props} />);
		expect(tree.getByTestId("details")).toBeTruthy();
	});

	it("has 1 child", () => {
		const tree = render(<Details route={props} />);
		//@ts-ignore
		expect(tree.toJSON().children.length).toBe(1);
	});
});
