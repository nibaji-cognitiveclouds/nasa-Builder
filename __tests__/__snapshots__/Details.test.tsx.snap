// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
  testID="details"
>
  <ActivityIndicator
    animating={true}
    color="red"
    hidesWhenStopped={true}
    size={20}
  />
</RCTSafeAreaView>
`;
